File: rev-com-server/src/main/java/com/codergb/bean/Director.java
package com.codergb.bean;

import lombok.*;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode(callSuper = true)
public class Director extends FileInfo{
  private String id;
  private String name;
  private String alias;
  private String avatarUrl;
  private String gender;
  private String birthPlace;
  private String profession;
  private String description;
  private Date createTime;
  private Date updateTime;
}

File: rev-com-manage/src/types/director.ts
import { IFileInfo } from "@/types/FileInfo";
import { IOccupation } from "@/types/occupation";
import { IMovie } from "@/types/movie";

interface IDirector extends IFileInfo {
  id: string;
  name: string;
  alias: string;
  avatarUrl: string;
  gender: string;
  birthPlace: string;
  occupations: IOccupation[];
  movies?: IMovie[];
  description: string;
  createTime: string;
  updateTime: string;
}
export type { IDirector };
